### DISCUSSION TITLE
Illegal instruction (core dumped)

### DISCUSSION THREAD

**Discussion by MaxKordt** [coqui-ai/TTS#1126] [General Q&A]
Hello,

for a project at my university I am currently developing a voice assistant app. I would like to use coqui TTS to synthesize speech. 

As a 'server' I got handed a Raspberry Pi 4. 
As os I installed a clean Ubuntu Server LTS and later ran `pip install TTS.` Worked so far. Now I am stuck when trying to use tts. 
I tried for example:
`tts-server --use_cuda=false --model_name tts_models/de/thorsten/tacotron2-DCA` or
`tts --use_cuda=false --text "Hello World!" --model_name tts_models/de/thorsten/tacotron2-DCA`

tts-server... results in:
/home/ubuntu/coquiTTS/lib/python3.8/site-packages/torch/package/_directory_reader.py:17: UserWarning: Failed to initialize NumPy: module compiled against API version 0xe but this version of numpy is 0xd (Triggered internally at  /root/pytorch/torch/csrc/utils/tensor_numpy.cpp:68.)
_dtype_to_storage = {data_type(0).dtype: data_type for data_type in _storages}
> tts_models/de/thorsten/tacotron2-DCA is already downloaded.
 > vocoder_models/de/thorsten/fullband-melgan is already downloaded.
 > Using model: Tacotron2
 > Model's reduction rate `r` is set to: 1
 > Vocoder Model: fullband_melgan
 > Generator Model: fullband_melgan_generator
 > Discriminator Model: melgan_multiscale_discriminator
 * Serving Flask app 'TTS.server.server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on all addresses.
   WARNING: This is a development server. Do not use it in a production deployment.
 * Running on http://[...]:5002/ (Press CTRL+C to quit)

After going to my browser and entering a the text "Hallo Welt!" is followed by:
 > Model input: Hallo Welt!
 > Text splitted to sentences.
['Hallo Welt!']
Illegal instruction (core dumped)

I have no clue what the Illegal instruction is, is caused by and how to resolve.
Since this is my first time working with Linux and anything related to TTS and coqui I tried following the warning and upgrade NumPy to get from 0xd to 0xe. But it is being aborted since TTS 0.5.0 needs this specific version of NumPy. Additionally I tried to downgrade pytorch but that does not even start the server and throw some other error. My python is 3.8.10

As I am now at my wits end what causes this and where to further search I hope someone knows this problem or has an idea in which area I need to search to fix this. 

Any input is much appreciated!

**Comment by fury88** [comment]
Did you ever figure this out? I'm having the same problem with Raspi 4. I've read it's possibly due to 64-bit OS vs. 32-bit?

**Comment by MaxKordt** [comment]
Hey,

it has been so long since I did this for an university project. I remember that I used coqui in my final work but on my laptop instead of a raspi. Used an old macbook. More eludes me... 

Wish you best of luck!

**Reply by fury88** [reply]
Thanks. Yeah its been a bit. LOL. I just installed Ubuntu and tried with no luck either. Must be the aarch64 chip in the Raspberry Pi board. Oh well
